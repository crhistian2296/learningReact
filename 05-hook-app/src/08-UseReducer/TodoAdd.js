import React from 'react';
import useForm from '../hooks/useForm';

const TodoAdd = ({ handleAdd }) => {
  const [{ description }, handleInputChange, reset] = useForm({ description: '' });

  const handleSubmit = (e) => {
    e.preventDefault();

    if (description.trim().length < 1) {
      return alert('Introduce your TODO before save');
    }
    const newTodo = {
      id: Date.now(),
      description,
      done: false,
    };

    handleAdd(newTodo);
    reset();
  };
  return (
    <div className='col-sm-5'>
      <h4>Agregar TODO</h4>
      <form action='todo' onSubmit={handleSubmit}>
        <input
          type='text'
          name='description'
          placeholder='Aprender...'
          className='form-control'
          onChange={handleInputChange}
          value={description}
        />
        <button type='submit' className='btn btn-outline-success mt-2'>
          Agregar
        </button>
      </form>
    </div>
  );
};

export default TodoAdd;
