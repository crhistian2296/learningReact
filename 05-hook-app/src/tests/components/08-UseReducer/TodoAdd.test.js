import { shallow } from 'enzyme';
import TodoAdd from '../../../08-UseReducer/TodoAdd';

describe('Pruebas en <TodoAdd/>', () => {
  const handleAdd = jest.fn();

  const wrapper = shallow(<TodoAdd handleAdd={handleAdd} />);

  test('Debe de mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('No debe de llamar a handleTodo si el valor TODO es vacio', () => {
    // * Forma alternativa mas conveniente si se produce submit
    // const formSubmit = wrapper.find('form').prop('onSubmit');
    // formSubmit({ preventDefault() {} });

    wrapper.find('button').simulate('click');
    expect(handleAdd).toHaveBeenCalledTimes(0);
  });

  test('Debe de llamar a handleTodo y resetear el campo despues', () => {
    const value = 'Aprender Typescript';
    wrapper.find('input').simulate('change', {
      target: {
        value,
        name: 'description',
      },
    });

    const formSubmit = wrapper.find('form').prop('onSubmit');
    formSubmit({ preventDefault() {} });

    expect(handleAdd).toHaveBeenCalledTimes(1);
    expect(handleAdd).toHaveBeenCalledWith({
      id: expect.any(Number),
      description: value,
      done: false,
    });
    expect(wrapper.find('input').prop('value')).toBe('');
  });
});
