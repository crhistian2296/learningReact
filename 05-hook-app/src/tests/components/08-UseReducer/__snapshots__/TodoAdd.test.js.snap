// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoAdd/> Debe de mostrarse correctamente 1`] = `
<div
  className="col-sm-5"
>
  <h4>
    Agregar TODO
  </h4>
  <form
    action="todo"
    onSubmit={[Function]}
  >
    <input
      className="form-control"
      name="description"
      onChange={[Function]}
      placeholder="Aprender..."
      type="text"
      value=""
    />
    <button
      className="btn btn-outline-success mt-2"
      type="submit"
    >
      Agregar
    </button>
  </form>
</div>
`;
